api/src/modules/propositions/propositions.handler.ts

```diff
@@ -0,0 +1,144 @@
+import type { FastifyReply as Reply, FastifyRequest } from "fastify";
+import type { proposition } from "@src/types/proposition";
+import type Filters from "@src/types/Filters";
+import filtersFactorySlots from "@src/utils/filtersFactorySlots";
+
+type Request = FastifyRequest<{
+  Querystring: {
+    filter: Filters.Slot;
+  };
+  Params: {
+    id: number;
+  };
+}>;
+
+export const handleGetAllSlots = async (request: Request, reply: Reply) => {
+  const { prisma } = request;
+  const filters = filtersFactorySlots(request.query.filter);
+
+  try {
+    const slots = await prisma.proposition_slot.findMany(
+      filters && {
+        where: { AND: [...filters] },
+      }
+    );
+
+    if (slots.length === 0) {
+      reply.code(404);
+      throw new Error("Aucun créneau disponible.");
+    }
+
+    reply.send(slots);
+  } catch (error) {
+    reply.send(error);
+  }
+}
+
+export const handleBookSlot = async (request: Request, reply: Reply) => {
+  const { prisma } = request;
+  const { id: slotId } = request.params;
+  const { id: userId } = request.user;
+
+  try {
+    const proposition: Array<proposition> = await prisma.$queryRaw`
+      SELECT * FROM pending_propositions WHERE user_id=${userId};
```

Comment:
I only react here but is this legit due to prisma ?
In standard this is a vulnerability for injections, but maybe Prisma sanitizes it under the hood ?

Reply:
Raw queries aren't deprecated. As we already have an SQL function that achieve what we need this makes sense to use it.

Reply:
This was not what I meant.

In a direct feed with the DB, this is an unprepared query and a vulnerability.
(remember how I did this with query objects with text and values properties).
I want to be sure prisma protect us against SQL injections in this case or if we have to "prepare" all of our raw queries.

Reply:
Oh right, maybe you should read Prisma documentation just to be safe. But as I understand it, Prisma take care of that for us.